tests/src/HolidaysTest.php

```diff
@@ -115,10 +115,42 @@ public function testAddDateTimeSpan()
         $this->assertEquals($expected, $holidays->getDates());
     }
 
+    public function dataExtendDateTimeSpan()
+    {
+        return [
+            [
+                new DateTimeSpan(new DateTime('2015-06-12'), new DateTime('2015-06-30')),
+                null,
+                new DateTimeSpan(new DateTime('2015-06-12'), new DateTime('2015-07-07')),
+            ],
+            [
+                new DateTimeSpan(new DateTime('2015-06-17'), new DateTime('2015-06-30')),
+                null,
+                new DateTimeSpan(new DateTime('2015-06-24'), new DateTime('2015-07-07')),
+            ],
+            [
+                new DateTimeSpan(new DateTime('2015-06-19'), new DateTime('2015-06-22')),
+                null,
+                new DateTimeSpan(new DateTime('2015-06-26'), new DateTime('2015-06-29')),
+            ],
+            [
+                new DateTimeSpan(new DateTime('2015-06-12'), new DateTime('2015-06-22')),
+                null,
+                new DateTimeSpan(new DateTime('2015-06-12'), new DateTime('2015-06-29')),
+            ],
+            [
+                new DateTimeSpan(new DateTime('2015-06-19'), new DateTime('2015-06-22')),
+                new DateTime('2015-06-18'),
+                new DateTimeSpan(new DateTime('2015-06-25'), new DateTime('2015-06-28')),
```

Comment:
If I'm getting this right, you should put another test where the start date is inside the span period.